### PR TITLE
Introduce Azure HttpObservable and SSL bypass properties

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spark/tools/http/OAuthTokenHttpObservable.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    public OAuthTokenHttpObservable(final OAuthTokenFetcher toGetAccessToken) {
+        super();
+
+        this.accessTokenFetcher = toGetAccessToken;
+        this.oauthDefaultRequestConfig = RequestConfig.custom()
+                .setCookieSpec(CookieSpecs.DEFAULT)
+                .build();
+    }
+
+    public OAuthTokenHttpObservable(final String accessToken) {
```
Comment (t-rufang) [inline src/main/java/com/microsoft/azure/spark/tools/http/OAuthTokenHttpObservable.java]: If we initialize the `OAuthTokenFetcher` in this way, how do we handle the token expire situation?

File: src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        cookieStore.addCookie(cookie);
+
+        this.httpMock.setCookieStore(this.cookieStore);
+    }
+
+    @Then("^send and check Http '(.+)' request to '(.*)' should contain headers$")
+    public void checkRequestHeaderAndParameters(String method,
+                                                String path,
+                                                Map<String, String> expectHeaders) {
+        this.httpMock.request(this.httpRequest, null, null, null ).subscribe();
```
Comment (t-rufang) [inline src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java]: Do we need to block this observable?

File: src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        expectHeaders.forEach((k, v) -> builder.withHeader(k, equalTo(v)));
+
+        verify(builder);
+    }
+
+    @Then("^send and check Https '(.+)' request to '(.*)' should contain headers$")
+    public void checkHttpsRequestHeaderAndParameters(String method,
+                                                String path,
+                                                Map<String, String> expectHeaders) {
+        this.httpMock.request(this.httpRequest, null, null, null ).subscribe();
```
Comment (t-rufang) [inline src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java]: Block this observable?

File: src/main/java/com/microsoft/azure/spark/tools/http/OAuthTokenHttpObservable.java (hunk 1 above)
Comment (wezhang) [reply src/main/java/com/microsoft/azure/spark/tools/http/OAuthTokenHttpObservable.java]: It's supposed `OAuthTokenFetcher` to handle token refresh in `get()` method

File: src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                "The exception caught was not a CertificateException: " + ExceptionUtils.getMessage(caught.get()),
+                caught.get().getCause() instanceof CertificateException);
+        assertEquals(expectErrorMessage, caught.get().getCause().getMessage());
+    }
+
+    @Then("send and check Https {string} request to {string} should throw SSLException {string}")
+    public void checkHttpsRequestSSLException(String method,
+                                                    String path,
+                                                    String expectErrorMessage) {
+        AtomicReference<Throwable> caught = new AtomicReference<>();
```
Comment (t-rufang) [inline src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java]: Just in curious, why do we need to make this filed `AtomicReference`

File: src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java (hunk 2 above)
Comment (wezhang) [reply src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java]: No need, the current scheduler is in the current thread

File: src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java (hunk 4 above)
Comment (wezhang) [reply src/test/java/com/microsoft/azure/spark/tools/http/HttpObservableScenario.java]: Since it's wrote in an lambda function. 😄

**Pull request merged**